unitTests/ui/StarRatingTest.ts

```diff
@@ -0,0 +1,155 @@
+import { l } from '../../src/Core';
+import { FakeResults } from '../Fake';
+import { $$ } from '../../src/utils/Dom';
+import * as Mock from '../MockEnvironment';
+import { IQueryResult } from '../../src/rest/QueryResult';
+import { StarRating, IStarRatingOptions } from '../../src/ui/StarRating/StarRating';
+
+const CONTAINER_CSS_CLASS = 'CoveoStarRating';
+const STAR_CSS_CLASS = 'coveo-star-rating-star';
+const ACTIVE_STAR_CSS_CLASS = 'coveo-star-rating-star-active';
+const STAR_LABEL_CSS_CLASS = 'coveo-star-rating-label';
+
+const getActiveStars = (starRatingElement: HTMLElement) => {
+  const children = starRatingElement.children;
+  let numStars = 0;
+  let numActiveStars = 0;
+  let i = 0;
+  for (; i < children.length; i++) {
+    if (children.item(i).classList.contains(STAR_CSS_CLASS)) {
+      if (children.item(i).classList.contains(ACTIVE_STAR_CSS_CLASS)) {
+        numActiveStars++;
+      }
+      numStars++;
+    }
+  }
+  return { numStars: numStars, numActiveStars: numActiveStars };
+};
+
+export function StarRatingTest() {
```

Comment:
Since it pulls data from index fields to display in the DOM, I'd add some test for XSS.

Eg: If the ratingField or numberOfRatings contains JS or HTML, it should still be safe.

Reply:
Also, could we start the function name with a lower-case char?

Reply:
every other test function starts with an upper-case though...

Reply:
Yes, the upper case is fine in this case :P